src/snapshot/tree.rs

```diff
@@ -137,6 +141,14 @@ impl RojoTree {
                         self.path_to_ids.insert(new_path.clone(), id);
                     }
                 }
+                if existing_metadata.specified_id != metadata.specified_id {
+                    if let Some(old) = &existing_metadata.specified_id {
+                        self.specified_id_to_refs.remove(old);
+                    }
+                    if let Some(new) = &metadata.specified_id {
+                        self.specified_id_to_refs.insert(new.clone(), id);
+                    }
+                }
```

Comment:
This needs a duplicated ID check like `RojoTree::insert_metadata`. Failing the check should probably result in an error message instead of a warning. Also, with the current implementation, a user specifying duplicate IDs can leave `specified_ids_to_refs` in an incorrect state:
```rust
                if existing_metadata.specified_id != metadata.specified_id {
                    // Adding, then removing a duped ID during a serve session 
                    // clobbers the preexisting mapping - this snippet should
                    // not run if we detect a duped ID
                    if let Some(old) = &existing_metadata.specified_id {
                        self.specified_id_to_refs.remove(old);
                    }
                    if let Some(new) = &metadata.specified_id {
                        self.specified_id_to_refs.insert(new.clone(), id);
                    }
                }
```

Reply:
Good catch, I'll add a check.